src/main/kotlin/nl/avisi/kotlinwebtest/jdbc/Jdbc.kt

```diff
@@ -0,0 +1,149 @@
+/**
+ * For licensing, see LICENSE.txt
+ * @author Martijn Heuvelink
+ */
+package nl.avisi.kotlinwebtest.jdbc
+
+import nl.avisi.kotlinwebtest.ExecutionContext
+import nl.avisi.kotlinwebtest.Executor
+import nl.avisi.kotlinwebtest.StepRequest
+import nl.avisi.kotlinwebtest.StepResponse
+import nl.avisi.kotlinwebtest.TestCase
+import nl.avisi.kotlinwebtest.TestStep
+import nl.avisi.kotlinwebtest.expressions.ExpressionEvaluator
+import nl.avisi.kotlinwebtest.expressions.findExpressions
+import org.slf4j.LoggerFactory
+import java.io.IOException
+import java.sql.DriverManager
+import java.sql.ResultSet
+import java.sql.SQLException
+import java.util.ArrayList
+import java.util.HashMap
+
+
+class JdbcStepRequest(var body: String? = null) : StepRequest, DatabaseRequest() {
+
+    override infix fun url(url: String) {
+        testStep.url = url
+    }
+
+    override infix fun driver(driver: String) {
+        testStep.driver = driver
+    }
+
+    lateinit var testStep: JdbcTestStep
+}
+
+class JdbcStepResponse(override val resultSet: List<HashMap<String, String>>?,
+                       override val success: Boolean,
+                       override val message: String? = null) : DatabaseStepResponse
+
+class JdbcTestStep(testCase: TestCase) : TestStep<JdbcStepRequest, JdbcStepResponse>(testCase, JdbcStepRequest()) {
+    var url: String? = null
+    var driver: String? = null
+
+    init {
+        request.testStep = this
+    }
+
+    fun resolveUrl(configuration: JdbcTestConfiguration): String? =
+            url ?: configuration.defaults.url
+
+    fun resolveDriver(configuration: JdbcTestConfiguration): String? =
+            driver ?: configuration.defaults.driver
+}
+
+class JdbcExecutor : Executor<JdbcTestStep> {
+
+    override fun execute(step: JdbcTestStep, executionContext: ExecutionContext): StepResponse {
+        val query = (step.request.query ?: error("No query configured for JDBC test step."))
+                .let { interpolateExpressions(it, executionContext) }
+
+        val configuration = executionContext.configuration[JdbcTestConfiguration::class]
+        val url = step.resolveUrl(configuration) ?: error("No url configured for JDBC test step.")
+        val driver = step.resolveDriver(configuration) ?: error("No driver configured for JDBC test step.")
+
+        var result: List<HashMap<String, String>>? = null
+
+        try {
+            Class.forName(driver)
+            log.info("Sending request:\n Querry: {}", query)
+            result = DriverManager.getConnection(url).use {
```

Comment:
This probably works for now. When there's a lot of JDBC calls going on we might need some kind of keep-alive or connection pooling.